src/phoenix/server/api/routers/auth.py

```diff
@@ -0,0 +1,187 @@
+)
+
+
+@router.post("/login")
+async def login(request: Request) -> Response:
+    data = await request.json()
+    email = data.get("email")
+    password = data.get("password")
+
+    if not email or not password:
+        raise HTTPException(status_code=HTTP_401_UNAUTHORIZED, detail="Email and password required")
+
+    async with request.app.state.db() as session:
+        user = await session.scalar(
+            select(OrmUser).where(OrmUser.email == email).options(joinedload(OrmUser.role))
+        )
+        if (
+            user is None
+            or (password_hash := user.password_hash) is None
+            or (salt := user.password_salt) is None
+        ):
+            raise HTTPException(status_code=HTTP_401_UNAUTHORIZED, detail=LOGIN_FAILED_MESSAGE)
+
+    loop = asyncio.get_running_loop()
+    password_is_valid = partial(
+        is_valid_password, password=password, salt=salt, password_hash=password_hash
+    )
+    if not await loop.run_in_executor(None, password_is_valid):
+        raise HTTPException(status_code=HTTP_401_UNAUTHORIZED, detail=LOGIN_FAILED_MESSAGE)
+
+    issued_at = datetime.now(timezone.utc)
+    refresh_token_claims = RefreshTokenClaims(
+        subject=UserId(user.id),
+        issued_at=issued_at,
+        expiration_time=issued_at + PHOENIX_REFRESH_TOKEN_MAX_AGE,
+        attributes=RefreshTokenAttributes(
+            user_role=UserRole(user.role.name),
+        ),
+    )
+    token_store: JwtStore = request.app.state.get_token_store()
+    refresh_token, refresh_token_id = await token_store.create_refresh_token(refresh_token_claims)
+    access_token_claims = AccessTokenClaims(
+        subject=UserId(user.id),
+        issued_at=issued_at,
+        expiration_time=issued_at + PHOENIX_ACCESS_TOKEN_MAX_AGE,
+        attributes=AccessTokenAttributes(
+            user_role=UserRole(user.role.name),
+            refresh_token_id=refresh_token_id,
+        ),
+    )
+    access_token, _ = await token_store.create_access_token(access_token_claims)
+    response = Response(status_code=HTTP_204_NO_CONTENT)
+    response = set_access_token_cookie(response, access_token)
+    response = set_refresh_token_cookie(response, refresh_token)
+    return response
+
+
+@router.post("/logout")
+async def logout(
+    request: Request,
+) -> Response:
+    token_store = request.app.state.get_token_store()
+    assert (user := request.user)
+    assert isinstance(user.identity, UserId)
+    user_id = int(user.identity)
+    token_ids = []
+    async with request.app.state.db() as session:
+        async with session.begin_nested():
+            for cls in (AccessTokenId, RefreshTokenId):
+                table = cls.table
+                stmt = delete(table).where(table.user_id == user_id).returning(table.id)
+                async for id_ in await session.stream_scalars(stmt):
+                    token_ids.append(cls(id_))
+    if token_ids:
+        await token_store.revoke(*token_ids)
```

Comment:
need to delete all tokens for the user for security reasons

Reply:
All tokens are being deleted.